### ISSUE TITLE
iOS. AlertDialog has a wrong layout if it is shown at the app startup

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-multiplatform#3540]
Compose 1.5.0-rc01, iOS

```
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.Button
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color

@Composable
fun App() {
    MaterialTheme {
        Box(Modifier.fillMaxSize().background(Color.White)) {
            var isDialogOpen by remember { mutableStateOf(true) }  // set to false to see how dialog should look
            Button(onClick = { isDialogOpen = true }) {
                Text("Open")
            }
            if (isDialogOpen) {
                androidx.compose.material3.AlertDialog(
                    onDismissRequest = { },
                    confirmButton = {
                        Button(onClick = { isDialogOpen = false }) {
                            Text("OK")
                        }
                    },
                    title = { Text("Alert Dialog") },
                    text = { Text("Lore ipsum") },
                )
            }
        }
    }
}
```
Showing at startup:
<img src="https://github.com/JetBrains/compose-multiplatform/assets/5963351/f4e4a719-321b-4307-b8ca-9a82f890a6ae" width="300">

Assumptions:
1. the initial contaizer size is zero, the initial density is 1f
2. the content doesn't react on the container size and density changes.

Showing at the button press (expected):
<img src="https://github.com/JetBrains/compose-multiplatform/assets/5963351/6d8631f5-a90e-4e60-ac7b-a12a65075dff" width="300">

Priority high, because we can have hidden races, and there can be some use cases (restoring state at the app startup)

**Comment by gastsail** [comment]
Is this the same if we add `modifier = Modifier.fillMaxWidth()` to the Texts inside the AlertDialog ?

**Comment by MatkovIvan** [comment]
It was fixed in `1.5.0-rc04`

**Comment by MatkovIvan** [comment]
It was combination of #3434 and #3778
